scene/main/node.cpp

```diff
@@ -1696,7 +1696,7 @@ TypedArray<Node> Node::find_children(const String &p_pattern, const String &p_ty
 			} else if (cptr[i]->get_script_instance()) {
 				Ref<Script> scr = cptr[i]->get_script_instance()->get_script();
 				while (scr.is_valid()) {
-					if ((ScriptServer::is_global_class(p_type) && ScriptServer::get_global_class_path(p_type) == scr->get_path()) || p_type == scr->get_path()) {
+					if ((ScriptServer::is_global_class(p_type) && ScriptServer::get_global_class_path(p_type) == scr->get_path()) || scr->get_path().contains(p_type)) {
```

Comment:
I don't think this is safe to do under any circumstances. You can match any unrelated script this way, just because it has a similar substring.

Reply:
Yes, that was a part of my point. It's not a corner case. It's a fairly major case, IMO.